src/components/Personalization/validateApplyPropositionsOptions.js

```diff
@@ -31,7 +31,7 @@ export default ({ logger, options }) => {
           objectOf({
             id: string().required(),
             schema: string().required(),
-            data: anything().required(),
+            data: objectOf(anything()),
```

Comment:
I think you need to do objectOf({}). But why change data to have to be an object?. It seems you could just remove this line entirely and allow data to be set or not.